README.md

```diff
@@ -1,38 +1,167 @@
+## Installation
 
+```
+ember install ember-sidenotes
+```
+
+## Usage
+
+This is a basic example of what this addon allow you to do.
+
+```hbs
+<SidenotesWrapper @items={{this.items}} as |Sidenote item|>
+  <Sidenote @id={{item.id}} @offsetY={{item.y}} @data={{item.data}} as |s|>
+    <p>
+      {{item.data.content}}
+    </p>
+    {{#if s.isActive}}
+      <button type='button' {{on 'click' s.unselect}}>Unselect</button>
+    {{else}}
+      <button type='button' {{on 'click' s.select}}>Select</button>
+    {{/if}}
+  </Sidenote>
+</SidenotesWrapper>
+```
+
+There are two ways of using this addon, controlled and non-controlled mode.
+
+## Controlled & Non-controlled modes
+
+### Controlled mode
 
-Installation
-------------------------------------------------------------------------------
+The controlled mode makes it possible to managed which side note is selected with an argument named `selectedSidenoteId`. This has to match the argument `id` passed to a component `<Sidenote @id={{...}} />`.
 
+As two actions `select` and `unselect` are exposed in each sidenote component, calling them will raise an action on the wrapper (respectively `onSelect` and `onUnselect`) with all the data necessary to be aware of which sidenote has been selected (`{ element, id, offsetY, data }`).
+
+Then, this is your job to mutate your argument `selectedSidenoteId` to make the new sidenote selected.
+
+_MyComponent.hbs_
+
+```hbs
+<SidenotesWrapper
+  @items={{this.items}}
+  @onSelect={{this.onSelect}}
+  @onUnselect={{this.onUnselect}}
+  @selectedSidenoteId={{this.selectedSidenoteId}}
+  as |Sidenote item|
+>
+  <Sidenote @id={{item.id}} @offsetY={{item.y}} @data={{item.data}} as |s|>
+    {{#if s.isActive}}
+      <button type='button' {{on 'click' s.unselect}}>Unselect</button>
+    {{else}}
+      <button type='button' {{on 'click' s.select}}>Select</button>
+    {{/if}}
+  </Sidenote>
+</SidenotesWrapper>
 ```
-ember install ember-sidenotes
+
+_MyComponent.js_
+
+```js
+export default class MyComponent extends Component {
+  @tracked items = [
+    { id: 1, offsetY: 104, data: null },
+    // ...
+  ];
+
+  @action
+  onSelect({ id }) {
+    this.selectedSidenoteId = id;
+  }
+
+  @action
+  onUnselect() {
+    this.selectedSidenoteId = null;
+  }
+}
 ```
 
+### Non-controlled mode
 
-Usage
-------------------------------------------------------------------------------
+The non-controlled mode is the way to use this addon in a simpler way. The selected sidenote will be handle internally by the components.
 
-[Longer description of how to use the addon in apps.]
+Both `select` and `unselect` actions will raised the matching action on the wrapper (respectively `onSelect` and `onUnselect`) but only for notify purpose.
 
+You can also specify a `defaultSidenoteId` argument to the wrapper to specify the sidenote selected on component load. If this is not specified or the one specified does not match any sidenote, this will default to the first sidenote.
```

Comment:
```suggestion
You can also specify a `defaultSidenoteId` argument to the wrapper to choose the sidenote selected on component load. If it's not specified or the one specified does not match any sidenote, this will default to the first sidenote.
```